theia-tree-editor/src/browser/master-tree-widget.tsx

```diff
@@ -308,13 +308,15 @@ export class MasterTreeWidget extends TreeWidget {
      * @param data The data array to generate the new tree nodes from
      * @param property The property of the parent data which will contain the new nodes.
      */
-    public addChildren(node: TreeEditor.Node, data: any[], property: string): void {
+    public async addChildren(node: TreeEditor.Node, data: any[], property: string): Promise<void> {
         const currentValue = node.jsonforms.data[property];
         let index = 0;
         if (Array.isArray(currentValue)) {
             index = currentValue.length;
         }
-        data.map((d, i) => this.nodeFactory.mapData(d, node, property, index + i));
+        for (const { d, i } of data.map((d, i) => ({ d, i }))) {
```

Comment:
Please re-write this to be more readable and avoid shadowing variable names: `data.map((d, i)` shadows `const { d, i }`.
E.g. something like this:
```
// d = data; i = index
const indexedData = data.map((d, i) => ({ d, i }));
for (const { d, i } of indexedData) {
  await this.nodeFactory.mapData(d, node, property, index + i);
}
```